tests/test_upload.py

```diff
@@ -620,6 +620,31 @@ def test_get_transactions_incorrect_totals(self, mock_logger):
 
         self.assertEqual(transactions, None)
 
+    @mock.patch('mtp_transaction_uploader.upload.get_authenticated_connection')
+    def test_excludes_records_from_other_accounts(self, mock_get_conn):
+        with open('tests/data/testfile_multiple_accounts') as f:
+            data_services_file = parse(f)
+
+        conn = mock_get_conn()
+        conn.batches.get.return_value = {
+            'count': 1,
+            'results': [{'id': 10}]
+        }
+
+        transactions = upload.get_transactions_from_file(data_services_file)
+
+        self.assertEqual(len(transactions), 2)
+        # transaction 0 - debit
```

Comment:
To clarify, does this correspond to https://github.com/ministryofjustice/money-to-prisoners-transaction-uploader/blob/5f2882156ec14043338bccd389f3e314c3104dfe/tests/data/testfile_multiple_accounts#L4 as this would seem to be the first `[DataRecord|BalanceRecord]` line in the file?

Reply:
yes exactly, i copied "testfile_1" to "testfile_multiple_accounts" and changed it to include transactions involving multiple accounts, but kept it in single-account format. there are 3 data records (which are individual credit/debit transactions) in both files. for the new file i did this with the data records:

1) kept it the same
2) changed the account number so it gets filtered out
3) changed the date for no real reason :)

Reply:
So the account number is the first field, padded by trailing zeros right? The amount forms the last non-zero digits, with some other numeric identifiers in between those two (account number and sort code)? Followed by a transfer reference containing the prisoners number and date of birth. After looking through bankline_parser I **think** I get what's going on here. The one thing I can't see is where `recieved_at` is encoded though?

Reply:
actually there are quite a lot of other fields in that line, [bankline-direct-parser](http://github.com/ministryofjustice/bankline-direct-parser) deals with parsing it. the first 6 characters are the sort code, the next 8 are the account number. then it gets complicated :) the transaction's date is actually `04036` in the sample which is a 2-digit year followed by a 3-digit day-of-year.